templates/systemd-override.erb

```diff
@@ -1,7 +1,8 @@
 <%- if scope.lookupvar('::osfamily') == 'Gentoo' -%>
 .include /usr/lib64/systemd/system/<%= @service_name %>.service
-<%- elsif scope.lookupvar('::operatingsystem') == 'Fedora' -%>
+<%- elsif scope.lookupvar('::operatingsystem') == 'Fedora' and scope.lookupvar('::operatingsystem') <= '31' -%>
 .include /lib/systemd/system/<%= @service_name %>.service
+<%- elsif scope.lookupvar('::operatingsystem') == 'Fedora' and scope.lookupvar('::operatingsystem') >= '32' -%>
```

Comment:
I think drop ins should work on all supported variants of the Red Hat OS family.
```suggestion
<%- elsif @facts['os']['family'] == 'Red Hat' -%>
```

In fact, I'm not sure if there's any systemd version that doesn't support this.

Reply:
I didn't want to break servers running older Fedora releases and anything older than F32 is EOL now which is why I limited this change to more modern releases.  I can run some tests on our Fedora 20 nodes to see if drop-in directories are supported but I didn't see anything in the documentation mentioning that feature.  

Another way to handle this would be to create a "systemd_version" fact which may work better than limiting options to just the RedHat OS family.

Reply:
It's supported on EL 7 which is based on Fedora 18ish IIRC. I wouldn't be surprised if it worked on even the ancient Fedora 17 that's listed in globals.pp.

Reply:
The oldest Fedora server I have available right now is running Fedora 20.  The man page for systemd.unit makes no mention of support for drop-in directories.  The man page on CentOS 7 *does* mention them which makes sense since EL7 runs systemd version 219 vs. 208 on Fedora 20.

Knowing this I should be able to make these changes work with Fedora 20 and up.

Reply:
@blackknight36 we have to include conditions for Redhat 8 since the deprecated warnings are coming there too.Will run the tests on all OSes and see if its break anything. Thanks alot for your contribution.

Reply:
Thanks, I forgot to add that.  I'll fix the template and push again.

Reply:
Ideally the include statement should be removed on *all* OSes, not just Redhat.  I'm sure that Gentoo also includes a newer version of systemd which would be affected in the future as well.  I don't have a Gentoo server to test on however.

Reply:
I think worries about drop ins are really not needed. This is literally the only reason I know that includes are a thing. On my Gentoo boxes I use drop ins regularly. I think that on all distros you can rely on them.